File: 2022/Day01/Day01.py
def getlist(lines) :
    res = []
    acc=0
    while len(lines) != 0 :
        if len(lines[0]) == 0 :
            res += [acc]
            acc=0
            lines.pop(0)
            continue
        acc+= int(lines[0])
        lines.pop(0)
    return res

with open('input.txt', 'r') as f :
    lines = [l.strip() for l in f.readlines()]

l = getlist(lines)
print(max(l))

l.sort()
l.reverse()
print(sum(l[:3]))

File: 2022/Day01/Day01.hs
module Main where
import Data.List

-- Get each line and creates a list of values
-- (list of the sums of snacks)
-- sets of values separated by blank lines
getList :: [String] -> Int -> [Int]
getList ("":l) acc = acc : (getList l 0)
getList (e:l) acc = getList l (acc + read e)
getList _ acc = acc :[]

main :: IO ()
main = do
    content <- readFile "input.txt"
    let input = lines content

    let res = getList input 0

    -- First exercise : Max
    print $ maximum res

    -- Second one, sum of the 3 tops
    print $ sum $ take 3 $ reverse $ sort res
